app/controllers/campaigns_controller.rb

```diff
@@ -50,6 +50,30 @@ def associate_seller_distributor
     json_response(campaign_json)
   end
 
+  # POST /campaigns/:id/gift_card
+  def generate_campaign_gift_cards
+    request_params = generate_campaign_gift_cards_params
+    unless @campaign.amount_raised > @campaign.amount_allocated + request_params['gift_card_amount_cents']
+      raise InvalidLineItem, "Request amount exceeds unallocated amount in campaign. Unallocated amount: #{
+        @campaign.amount_raised - @campaign.amount_allocated
+      }"
+    end
+
+    WebhookManager::GiftCardCreator.call(
+      {
+        amount: request_params['gift_card_amount_cents'],
+        single_use: true,
+        project_id: @campaign.project_id,
+        campaign_id: @campaign.id,
+        distributor_id: @distributor.id,
+      }
+    )
+
+    json_response({
+      'amount_left_to_allocate': @campaign.amount_raised - @campaign.amount_allocated
```

Comment:
Can we add this number to the Campaign GET

Reply:
sure i can do that in a follow up